api/config/v1alpha1/envoyproxy_types.go

```diff
@@ -18,8 +23,21 @@ type EnvoyProxy struct {
 
 // EnvoyProxySpec defines the desired state of EnvoyProxy.
 type EnvoyProxySpec struct {
```

Comment:
Just refreshed my memory by going through https://github.com/envoyproxy/gateway/blob/main/docs/design/CONFIG_API.md#data-plane-api

based on the agreed upon design, the `EnvoyGateway` config, passed via the cmdline arg,  decides the configuration for the control plane components/runners/services
and the `EnvoyProxy` resources passed into K8s, linked to the `GatewayClass` decides the configuration of the dataplane Envoy proxy fleet.

The Xds Server address & port (not included here but is required) is needed by the proxy to connect to the server and the same address (optional, can bind to all addresses) and port is required by the server to listen for xds client connections, so such a configuration is needed in `EnvoyGateway` too and the values in both must match .
It has been hardcoded in the `xds-server` for now https://github.com/envoyproxy/gateway/blob/d99f32feef8ddd6ede45bd0c93b607586138d4c8/internal/xds/server/runner/runner.go#L57

Reply:
@arkodg agreed that we should have logic to ensure the EnvoyProxy xds config aligns with how EG is running. IMO we need to be careful to not outsmart ourselves as different client/server config permutations will work. I think we track this as an issue to improve in the future. In the meantime, the API docs should suffice or I can elaborate on the docs if needed. THoughts @arkodg @LukeShu @youngnick @skriss